lib/fine-types/app/Options/Common.hs

```diff
@@ -0,0 +1,45 @@
+{-# LANGUAGE ApplicativeDo #-}
+{-# LANGUAGE DuplicateRecordFields #-}
+{-# LANGUAGE LambdaCase #-}
+
+module Options.Common where
+
+import Prelude
+
+import Options.Applicative
+    ( Parser
+    , ReadM
+    , eitherReader
+    , help
+    , long
+    , metavar
+    , option
+    , short
+    , value
+    )
+
+inputOption :: Parser (Maybe FilePath)
+inputOption =
+    option fileReader
+        $ mconcat
+            [ long "input"
+            , short 'i'
+            , metavar "FILE"
+            , help "input file (stdin if not specified or --)"
```

Comment:
Prompted by this pull request, I searched the internet and it appears that the canonical method to use `stdin` in lieu of a filename is to use the special file [`/dev/stdin`][1]. 🤔 I leave it up to you whether to change that, strictly speaking, that would be outside the scope of this pull request.

  [1]: https://serverfault.com/q/398514

Reply:
I was going to change it and remove the Maybe but it looks like `/dev/stdin` is not a Windows thing while pipes should work
I've also found out that I meant `-` not `--` , but I will remove it entirely